app/controllers/campaigns_controller.rb

```diff
@@ -98,15 +108,21 @@ def name_and_id
 
   def peer_to_peer
     session[:donor_signup_url] = request.env["REQUEST_URI"]
-    @npo = Nonprofit.find_by_id(params[:npo_id])
+    @nonprofit = Nonprofit.find_by_id(params[:npo_id])
+    @parent_campaign = Campaign.find_by_id(params[:campaign_id])
+    @child_campaign = Campaign.where(
+      profile_id: current_user.profile.id,
```

Comment:
When the user goes here, they'll probably be logged out. If so, the page crashes here as current_user is nil.